compiler+runtime/src/cpp/jank/util/clang.cpp

```diff
@@ -234,24 +232,62 @@ namespace jank::util
       return "/virtual/incremental.pch";
     }
 
-    auto dev_path{ jank_path / "incremental.pch" };
-    if(std::filesystem::exists(dev_path))
-    {
-      return dev_path.c_str();
-    }
-
     std::string const installed_path{ format("{}/incremental.pch",
                                              user_cache_dir(binary_version)) };
     if(std::filesystem::exists(installed_path))
     {
       return installed_path.c_str();
     }
-
     return none;
   }
 
+  /* In order to allow the user of jank to be guaranteed access to specific system
+   * libraries at runtime in their program, we allow them to customize the 
+   * pre-compiled header, the template for which, is the prelude.hpp file. The flag
+   * by which they do this is --include-pch. This function takes those inputs, such as
+   * "float.h" or "cfloat" and appends them as #include directives to a copy of the prelude.hpp file. 
+   * This ensures that when the user includes these headers in their program, they will be available. 
+   * It prevents the user from depending on transitive dependencies included by jank itself as a basis 
+   * for writing jank programs. 
+   * 
+   * Given the original prelude.hpp path, the pch_includes from the CLI opts, and the binary version
+   * Return the filesystem path of the customized prelude header file.
+   */
+  static std::filesystem::path
+  customize_prelude(std::filesystem::path const &prelude_path,
+                    native_vector<jtl::immutable_string> const &pch_includes,
+                    jtl::immutable_string const &binary_version)
+  {
+    /* Copy the prelude.hpp template to a temporary directory */
+
+    std::filesystem::path customized_prelude_path = format("{}/jank_prelude_customized_{}.hpp",
+                                                           std::filesystem::temp_directory_path(),
+                                                           binary_version)
+                                                      .c_str();
+    auto const copy_options = std::filesystem::copy_options::overwrite_existing;
```

Comment:
Please always use direct initialization. i.e.

```cpp
auto const copy_options{ std::filesystem::copy_options::overwrite_existing };
```

Please do this for the entire PR (I won't comment on any more of them this review).